tests/zstreamtest.c

```diff
@@ -1028,6 +1028,55 @@ static int basicUnitTests(U32 seed, double compressibility)
     }
     DISPLAYLEVEL(3, "OK \n");
 
+    /* Large source and large job size multithreading compression test */
```

Comment:
how long does this test last ?

Reply:
Less than 10 seconds

Reply:
It's very reasonable time duration to test such condition,

but for _this section of the test system_ (`basicUnitTests()` in `zstreamtest`), it might be a bit too long :
we ideally want all tests in `basicUnitTests()` to be a fraction of a second each,
so that the sum of all these tests remain in the "acceptable" range to be run as a preamble test.

Any chance this could become an equivalent test in the `--test-large-data` of `playTests.sh` ?
That's generally where we play all long tests with large files, making it possible to disable them selectively.

One downside is that `playTests.sh` uses the CLI, rather than the API.
On first look, it seems it could nonetheless be compatible with below test ?

Reply:
We can't control the job size on the CLI except implicitly. I also have a test there that exposes the bug.

I think we should leave the test there for now, because we don't currently have a better place.

But, I think we should invest some time in switching to gtest for our unit tests. It is the standard for C/C++ tests. There are a couple of benefits:

* All the tests can start with a clean environment and share setup/teardown code. We can reset or recreate contexts between tests, populate input buffers, etc.
* gtest has great helpers for printing why a test failed.
* We can do fancy things like run the same test on multiple types of data (if we want).
* When one test fails it still runs all the other tests
* It is the standard, so on-boarding engineers and open source contributors will be easier.

The downside is that we have a C++/cmake dependency for testing (gtest can be a submodule). I think that is reasonable since it is only needed for tests.

Reply:
Agreed !

Reply:
I ended up removing the test because it is failing in 32-bit mode with ASAN.

When we introduce gtest we'll have to skip the "expensive" tests in 32-bit ASAN mode like we do with playTests.sh.